Server/src/models/DatasetModels/DatasetDeleteModel.ts

```diff
@@ -0,0 +1,276 @@
+        let isDatasetDataTypeInUse = await this.selectOneUseOfDatasetDataTypeQuery(datasetDataTypeId)
+        if (isDatasetDataTypeInUse == undefined) {
+            await this.deleteDatasetDataTypeQuery(datasetDataTypeId)
+        }
+    }
+
+    async deleteAuthorsOfPublication(publicationsId: number) {
+        let isAuthorInUse: any
+        let authorsToDelete: number[] = []
+        let rawAuthorIds = await this.connection.query("SELECT authorsId FROM publications_authors_authors WHERE publicationsId = ?", [publicationsId])
+        await this.connection.query("DELETE FROM publications_authors_authors WHERE publicationsId = ?", [publicationsId])
+        // Check which authors are still in use
+        for (let index = 0; index < rawAuthorIds.length; index++) {
+            isAuthorInUse = await this.connection.query("SELECT publicationsId FROM publications_authors_authors WHERE authorsId = ?", [rawAuthorIds[index].authorsId]);
+            if (isAuthorInUse.length < 1) {
+                authorsToDelete.push(rawAuthorIds[index].authorsId)
+            }
+        }
+        if (authorsToDelete.length > 0)
+            await this.deleteAuthorsQuery(authorsToDelete)
+    }
+
+    async deletePublication(publicationId: number) {
+        let isPublicationInUse = await this.selectOneUseOfPublicationQuery(publicationId)
+        if (isPublicationInUse == undefined) {
+            await this.deletePublicationQuery(publicationId)
+        }
+    }
+    async deletePublisher(publisherId: number) {
+        let isPublisherInUse = await this.selectOneUseOfPublisherQuery(publisherId)
+        if (isPublisherInUse == undefined) {
+            await this.deletePublisherQuery(publisherId)
+        }
+    }
+
+    async deletePublicationType(publicationTypeId: number) {
+        let isPublicationTypeInUse = await this.selectOneUseOfPublicationTypeQuery(publicationTypeId)
+        if (isPublicationTypeInUse == undefined) {
+            await this.deletePublicationTypeQuery(publicationTypeId)
+        }
+    }
+
+    async rejectDataset(id: number): Promise<string> {
+        let rawDatasetFKs = await this.commonModel.selectDatasetFKQuery(id);
+        // First clear tables that need dataset ID
+        await this.commonModel.deleteDatapointCommentsQuery(id)
+        await this.deleteDataPointsOfDataset(id)
+        await this.deleteMaterialsOfDataset(id)
+        // Delete dataset proper
+        await this.commonModel.wipeEntryFromUnapprovedTable(id)
+        await this.deleteDatasetQuery(id)
+        await this.deleteDatasetDataType(rawDatasetFKs.datatypeId)
+        // Delete publication and its data
+        await this.deleteAuthorsOfPublication(rawDatasetFKs.publicationId)
+        await this.deletePublication(rawDatasetFKs.publicationId)
+        await this.deletePublisher(rawDatasetFKs.publisherId)
+        await this.deletePublicationType(rawDatasetFKs.publicationTypeId)
+        return "Successfully deleted data set"

```

Comment:
maybe change to "Successfully removed data set."? deleted seems... harsh

Reply:
very minor + this is backend, consumers are literally executing a delete request, the FE will not display the text the backend supplies on success

Reply:
Ahh okay, thought this is what shows when the admin rejects a dataset

Reply:
Ideally, I am returning responses to eventually move them into loggers

Reply:
Changed it to removed